### DISCUSSION TITLE
Using libavif instead of libheif to write AVIF files

### DISCUSSION THREAD

**Discussion by dooman87** [ImageMagick/ImageMagick#4369] [Development] [answered]
Hi,

Over the time I'm having more and more examples of poor AVIF encoding that results with the images with a lot of visible artifacts. I created issues (one example here: https://github.com/strukturag/libheif/issues/407) for some of those. I used the latest libheif and libaom versions in my experiments.

This morning, out of curiosity I tried libavif and results look much better (no artifacts) and the sizes are even smaller in some cases. I used the same `libaom` library as a backend. Unfortunately, I don't have knowledge to fix libheif, but I should be able to create a new AVIF coder or update `heic.c` one with the option to use `libavif` when writing AVIF.  Would you consider such a change if I'll make it and create PR? 

The example of encoded images (can upload more if needed):
[avif-example.zip](https://github.com/ImageMagick/ImageMagick/files/7349154/avif-example.zip)

**Comment by urban-warrior** [comment, marked as answer]
ImageMagick at some point supported libavif but we ended up with libheif instead.  The libheif delegate library is under development and improving over time.  Posting any issues you have with libheif to the developers is the right thing to do.  As they fix issues, the ImageMagick community benefits.  Given ImageMagick is open-source, you certainly can add a coder to your instance of ImageMagick to support AVIF with the libavif delegate library.
